mission_3/routes/products.js

```diff
@@ -0,0 +1,335 @@
+import express from 'express';
+import { PrismaClient } from '@prisma/client';
+
+const router = express.Router();
+const prisma = new PrismaClient();
+
+
+
+async function validateProduct (req, res, next) {
+  const {
+    name,
+    description,
+    price,
+    tags
+  } = req.body;
+
+  // validation logic (possible improvements with Zod library in the future)
+  if ( !name || !description || !price || !tags )
+    return res.status(400).json({ message: "Invalid SQL Parameters" });
+
+  next();
+}
+
+
+
+router.route('/')
+
+  // Upload a new product
+  .post(validateProduct, async (req, res) => {
+    // destructuring field data from request body
+    const {
+      name,
+      description,
+      price,
+      tags
+    } = req.body;
+
+    try {
+      // insert into Product table
+      const product = await prisma.product.create({
+        data: {
+          name,
+          description,
+          price,
+          tags
+        }
+      });
+
+      // send 201 response status code indicating the query was processed successfully
+      // with the Product record
+      res.status(201).json(product);
+    } catch (err) {
+      console.error('An error has occurred: ', err);
+
+      res.status(500).json({ message: "An error has occurred during processing sql" });
+    }
+  })
+
+  // Inquiry all products
+  .get(async (req, res) => {
+    const { 
+      offset = 0, 
+      limit = 10, 
+      sort = 'recent',
+      search = ''
+    } = req.query;
+
+    let productsSort = '';
+
+    switch (sort) {
+      case 'old':
+        productsSort = 'asc';
+        break;
+
+      case 'recent':
+      default:
+        productsSort = 'desc';
+    }
```

Comment:
아래와 같이 작성하면 더 가독성있지 않을까요? 실제로 코드를 작성하다보면 let 보다는 주로 const를 사용하게 됩니다. const를 주력으로 사용하는걸 연습해보세요!

```suggestion
    let productsSort = 'desc';

    if (sort === 'old') {
      productsSort = 'asc'
    }
```

혹은


```suggestion
    const productsSort = sort === 'old' ? 'asc' : 'desc';
```